remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+
+
+oldgenome=''
+newgenome=''
+newgenomeaccession=''
+vcffile=''
+outfile=''
+
+while getopts 'g:n:a:v:o:' flag; do
+	case "${flag}" in
+		g) oldgenome="${OPTARG}";;
+		n) newgenome="${OPTARG}" ;;
+		a) newgenomeaccession="${OPTARG}" ;;
+		v) vcffile="${OPTARG}" ;;
+		o) outfile="${OPTARG}" ;;
+	esac
+done
+
+
+#1) Flanking sequence generation
+echo "-----------------------1) Flanking sequence generation-----------------------"
+#generate chrom sizes:
+samtools faidx $oldgenome 
+cut -f1,2 "$oldgenome".fai > "$oldgenome".chrom.sizes
+
+#filter SNPs only
+bcftools filter -i 'TYPE="snp"' $vcffile > snps_only.vcf
+
+#store header
+awk '$1 ~ /^#/ {print $0}' snps_only.vcf > vcf_header.txt
+
+#convert vcf to bed:
+vcf2bed < snps_only.vcf > variants.bed
+#the actual position of the variant is the second coordinate
+rm snps_only.vcf
+
+#generate the flanking sequence intervals
+bedtools flank -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
+rm variants.bed "$oldgenome".chrom.sizes
+
+#this outputs a flanking.bed file: for 1 variant we have the following lines:
+#[chr] [-50 pos] [variant pos-1] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+#[chr] [variant pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#we want 1 line per variant, with 1 interval to create the read:
+#[chr] [-50 pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#fix rows: put both flanking sequences on 1 row so each read is 101 bases: [50 bases]-[variant]-[50 bases]
+awk 'NR == 1 {prev = $2} NR>1 && NR%2==0 {print $1"\t"prev"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9} NR%2 == 1 { prev = $2}' flanking.bed > flanking.corrected.bed
+rm flanking.bed
+
+#get the fasta sequences for these intervals
+bedtools getfasta -fi $oldgenome -bed flanking.corrected.bed -fo variants_reads.fa
+
+#replace the colon separators with "|"
+sed -i 's/:/|/' variants_reads.fa
+
+#store rsIDs
+awk '{print $4}' flanking.corrected.bed > rsIDs.txt
+
+#store variant bases
+awk '{print $7}' flanking.corrected.bed > variant_bases.txt
+
+#store the other vcf columns
+awk '{print $5, $8, $9}' flanking.corrected.bed > qual_filt_info.txt
+
+#paste the names, variant bases, then fasta sequences into a new file
+paste <(grep '^>' variants_reads.fa) variant_bases.txt rsIDs.txt qual_filt_info.txt <(grep -v '^>' variants_reads.fa) > temp.txt
+rm flanking.corrected.bed variant_bases.txt rsIDs.txt qual_filt_info.txt  variants_reads.fa
+
+#reformat the fasta ID
+sed 's/\t/|/g; s/ /|/g; s/\(.*\)|/\1 /' temp.txt | tr ' ' '\n' > variant_reads.out.fa
+rm temp.txt
+
+
+
+
+
+
+#2) Mapping with bowtie2
+echo "---------------------------2) Mapping with bowtie2---------------------------"
+#"index" is the prefix of the output filenames, not a directory
+echo "#### Building index: may take a while ####"
+bowtie2-build $newgenome index
+echo "#### Mapping results: ####"
+bowtie2 -f -x index variant_reads.out.fa | samtools view -bS - > reads_aligned.bam
```

Comment:
1) Could you please provide an explanation of the `bowtie21 flags used?
2) Although we have observed that Bowtie only stores one alignment per “read”, it's better to specify this behaviour explicitly (using flags)